src/openlifu/vf/virtual_fit.py

```diff
@@ -0,0 +1,161 @@
+import logging
+from dataclasses import dataclass, field
+from typing import Optional, Tuple
+
+import numpy as np
+import xarray as xa
+
+from openlifu.geo import Point
+from openlifu.plan import TargetConstraints
+from openlifu.xdc import Transducer
+
+
+@dataclass
+class VirtualFit:
+    """
+    VirtualFit class.
+
+    Represents the virtual fitting algorithm which consists in
+    finding the optimal transducer transform (position and orientation)
+    given an input MRI volume and the associated target.
+    """
+    pitch_range: Tuple[int, int] = (10, 40)
+    """The pitch range for the grid search."""
+
+    pitch_step: int = 3
+    """The pitch step for the grid search."""
+
+    yaw_range: Tuple[int, int] = (-5, 25)
+    """The yaw range for the grid search."""
+
+    yaw_step: int = 3
+    """The yaw step for the grid search."""
+
+    search_range_units: str = "deg"
+    """Search grid units."""
+
+    steering_limits: Tuple[TargetConstraints] = field(default_factory=list)
+    """Defines the accepteable range for a target in the transducer space, usually (lat, ele, ax)."""
```

Comment:
Should these properties, from `pitch_range` to `steering_limits`, be properties of the Transducer? Or of the Protocol? I feel like it's one of those... if agreed we can open an issue for later putting them in Transducer or Protocol. @peterhollender any opinion?

Reply:
pitch and yaw ranges should come from the Protocol, as they are determined by the target's general location

Reply:
Makes sense. And steering limits are more of a Transducer thing right? Like we would do some simulations to determine steering limits for each Transducer and then hardcode those limits into the Transducer definition

Reply:
If that is the case, then we would probably remove those attributes from `VirtualFit`, we don't want to duplicate those on disk (the json file associated to `VirtualFit`). For now I will leave them for simplicity, and we can sync about that before merging.